test/test_magnet.cpp

```diff
@@ -476,3 +476,82 @@ TORRENT_TEST(invalid_web_seed_escaping)
 	TEST_CHECK(ec);
 }
 
+TORRENT_TEST(parse_magnet_select_only)
+{
+	error_code ec;
+	add_torrent_params p = parse_magnet_uri("magnet:?xt=urn:btih:cdcdcdcdcdcdcdcdcdcdcdcdcdcdcdcdcdcdcdcd"
+		"&dn=foo&so=0,2,4,6-8", ec);
+	TEST_CHECK(!ec);
+
+	TEST_EQUAL(p.file_priorities.size(), 9);
+	for (std::size_t i = 0; i < p.file_priorities.size(); i++)
+	{
+		if (i == 0 || i == 2 || i == 4 || i == 6 || i == 7 || i == 8)
+		{
+			TEST_EQUAL(p.file_priorities[i], default_priority);
+		}
+		else
+		{
+			TEST_EQUAL(p.file_priorities[i], dont_download);
+		}
+	}
+
+	p = parse_magnet_uri("magnet:?xt=urn:btih:cdcdcdcdcdcdcdcdcdcdcdcdcdcdcdcdcdcdcdcd"
```

Comment:
if it's not too much trouble, I think these should be separated into individual test cases. If something breaks, it's a lot easier to pin-point which case broke in that case

Reply:
OK